src/key_values/error.rs

```diff
@@ -0,0 +1,50 @@
+use std::fmt;
+
+/// An error encountered while working with structured data.
+#[derive(Clone, Debug)]
+pub struct KeyValueError {
+    msg: &'static str,
+}
+
+impl KeyValueError {
+    /// Create an error from the given message.
+    pub fn msg(msg: &'static str) -> Self {
+        KeyValueError {
+            msg: msg,
+        }
+    }
+}
+
+impl fmt::Display for KeyValueError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.msg.fmt(f)
+    }
+}
+
+impl From<fmt::Error> for KeyValueError {
+    fn from(_: fmt::Error) -> Self {
+        KeyValueError::msg("formatting failed")
+    }
+}
+
+impl From<KeyValueError> for fmt::Error {
+    fn from(_: KeyValueError) -> Self {
+        fmt::Error
+    }
+}
+
+#[cfg(feature = "std")]
+mod std_support {
+    use super::*;
+    use std::error;
+
+    impl error::Error for KeyValueError {
+        fn description(&self) -> &str {
+            self.msg
+        }
+
+        fn cause(&self) -> Option<&error::Error> {
+            None
+        }
+    }
+}
```

Comment:
Could you see whether you can set your editor to end every line with a newline? Without that, this isn't a "text file" as defined by POSIX and behaves badly with some cli commands.

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline